examples/GainPlugin/PluginProcessor.cpp

```diff
@@ -186,10 +194,25 @@ AudioProcessorEditor* GainPluginAudioProcessor::createEditor()
     editor->setResizeLimits(400, 240, 400 * 2, 240 * 2);
     editor->getConstrainer()->setFixedAspectRatio(400.0 / 240.0);
     editor->setSize (400, 240);
+    
+    // Get appRoot reference created by GenericEditor in order to dispatch events
+    appRoot = &editor->getReactAppRoot();
```

Comment:
This is a bit dangerous here as once we hand over the pointer to our GenericEditor to the caller, we have no control over the lifetime of that object. The host (e.g. a DAW) can close the editor at will and totally invalidate this pointer, which we've saved here and call in the timer callback.

The safer move would be this:
* Don't hang onto the `appRoot` pointer here.
* In the timer callback, we use `getActiveEditor` to see if we still have an editor alive: https://docs.juce.com/master/classAudioProcessor.html#a0f1386123ab008b42a82c7d121b2ff13

Then `timerCallback` can look more like this:

```
if (auto* editor = dynamic_cast<reactjuce::GenericEditor*>(getActiveEditor())
{
    // Here we know we have a safe handle to the editor
    editor->getReactAppRoot().dispatchEvent(
        "gainPeakValues",
        static_cast<float>(gainPeakValue),
        static_cast<float>(gainPeakValue)
    );
}

Reply:
Thanks! That was challenging bit I wasn't sure about. Now clear!